vyper/semantics/types/subscriptable.py

```diff
@@ -263,6 +263,9 @@ def from_annotation(cls, node: vy_ast.Subscript) -> "DArrayT":
 
         value_type = type_from_annotation(node.slice.value.elements[0])
 
+        if isinstance(value_type, HashMapT):
```

Comment:
this should use `if value_type._as_array`, which mirrors the implementation for SArrayT.